lib/payday/invoiceable.rb

```diff
@@ -20,18 +20,22 @@ def bill_to
 
   # Calculates the subtotal of this invoice by adding up all of the line items
   def subtotal
-    line_items.reduce(BigDecimal.new("0")) { |result, item| result += item.amount }
+    line_items.reduce(BigDecimal.new("0")) { |result, item| result += item.amount_no_tax }
+  end
+
+  def items_tax
+    line_items.reduce(BigDecimal.new("0")) { |result, item| result += item.item_tax }
+  end
+
+  def subtotal_with_items_tax
+    subtotal + items_tax
   end
 
   # The tax for this invoice, as a BigDecimal
   def tax
-    if defined?(tax_rate)
-      calculated = subtotal * tax_rate
-      return 0 if calculated < 0
-      calculated
-    else
-      0
-    end
+      calculated = subtotal_with_items_tax * tax_rate
```

Comment:
@alonhartal I've reviewed this one more time, it's very not clear.
In general, there are 2 possible taxes in payday - per line and general tax rate. In any case you should not use both, we use tax per line only. I think I will remove all this logic from tax that related to general tax_rate, it's confusing and no purpose for this. What do u think?

Reply:
makes sense for me @katiaAlexandrov